### PR TITLE
Vaibhav/another sanity check

### PR DESCRIPTION
**Pull request by vvalecha519**
-Includes Sanity Checks
-Fixed Issue if config file (db) does not exist
-Prevent overwriting keystore for indices that exist
-Added check if mnemonics exists since failing test

Screenshot 2024-05-27 at 3 32 33 PM

### CODE REVIEW

File: src/electron/utils/storage.js [hunk 1]
```diff
@@ -92,7 +93,7 @@ class Database {
 
     getAllStakerAddresses(dbPassword) {
         const allStakerAddress = this._store.get("stakerAddress");
-
+        if (!allStakerAddress) return allStakerAddress;
```
Comment (seongyun-ko) [inline src/electron/utils/storage.js:96]: return UNDEFINED;

File: src/electron/listeners.js [hunk 2]
```diff
@@ -216,14 +216,21 @@ const genValidatorKeysAndEncrypt = async (
         withdrawalAddr2StakeInfo[info.withdrawalSafeAddress].push(info)
     }
 
+    for (const [withdrawalSafeAddress, infoArray] of Object.entries(withdrawalAddr2StakeInfo)) {
+        const ids = infoArray.map((i) => i.validatorID)
+        validatorIDs.push(...ids)
+    }
+    const isUnique = storage.checkUniqueValidatorIndices(validatorIDs, address)
```
Comment (seongyun-ko) [inline src/electron/listeners.js:223]: these are `validator ID` which is internal representation of validator within ether.fi

`validator index` is a reserved term to refer their representation in beacon layer.

**Review by seongyun-ko — APPROVED** [review]
LGTM, just the variable naming can be improved. can you leave some comments related with it so that peope don't get wrong

**Review by seongyun-ko — APPROVED** [review]
lgtm

File: src/electron/utils/storage.js [hunk 3]
```diff
@@ -54,6 +58,49 @@ class Database {
...
+    }
+
+    async checkPhasesOfCollidingIds(collidingIds) {
+        console.log("checking phases of colliding ids", collidingIds)
+        const ALCHEMY_API_KEY = 'Wo1ZTJHKM_EqFK5Q7weYMNeeR42VShY3'
+        const ALCHEMY_URL = `https://eth-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`; 
+        const NODE_MANAGER_ADDRESS = '0x8b71140ad2e5d1e7018d2a7f8a288bd3cd38916f';
+        const web3=new Web3(ALCHEMY_URL);
+        const validatorsInWrongPhase = []
+        const contract = new web3.eth.Contract(ABI, NODE_MANAGER_ADDRESS);
```
Comment (tamebadger) [inline src/electron/utils/storage.js:88]: can we move this out to not be initialized every time this function runs ? should speed things up quick a bit

**Comment by seongyun-ko** [comment]
LGTM!

**Review by azocardanny12-lgtm — APPROVED** [review]

**Pull request merged**